File: src/main/kotlin/org/tera201/vcsmanager/scm/RepositoryFile.java
package org.tera201.vcsmanager.scm;

import org.tera201.vcsmanager.util.RDFileUtils;

import java.io.File;

public class RepositoryFile {

	private File file;

	public RepositoryFile(File file) {
		this.file = file;
	}

	public File getFile() {
		return file;
	}

	public boolean fileNameEndsWith(String suffix) {
		return file.getName().toLowerCase().endsWith(suffix.toLowerCase());
	}

	public boolean fileNameMatches(String regex) {
		return file.getName().toLowerCase().matches(regex);
	}

	public String getFullName() {
		return file.getAbsolutePath();
	}

	public boolean fileNameContains(String text) {
		return file.getName().toLowerCase().contains(text);
	}
	
	public String getSourceCode() {
		return RDFileUtils.readFile(getFile());
	}
	
	@Override
	public String toString() {
		return "[" + file.getAbsolutePath() + "]";
	}
	
	
	
}

File: src/main/kotlin/org/tera201/vcsmanager/scm/RepositoryFile.kt
package org.tera201.vcsmanager.scm

import org.tera201.vcsmanager.util.RDFileUtils.readFile
import java.io.File
import java.util.*

class RepositoryFile(val file: File) {
    fun fileNameEndsWith(suffix: String): Boolean {
        return file.name.lowercase(Locale.getDefault()).endsWith(suffix.lowercase(Locale.getDefault()))
    }

    fun fileNameMatches(regex: String): Boolean {
        return file.name.lowercase(Locale.getDefault()).matches(regex.toRegex())
    }

    val fullName: String
        get() = file.absolutePath

    fun fileNameContains(text: String): Boolean {
        return file.name.lowercase(Locale.getDefault()).contains(text)
    }

    val sourceCode: String
        get() = readFile(file)

    override fun toString(): String {
        return "[" + file.absolutePath + "]"
    }
}
